wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -187,9 +185,13 @@ public void configurePipeline(PipelineConfigurer configurer) {
           directives = String.format("#pragma load-directives %s;%s", config.getUDDs(), config.getDirectives());
         }
       }
-
+      if (LANGUAGE_SQL.equalsIgnoreCase(config.getExecutionLanguage())
+              && Strings.isNullOrEmpty(config.preconditionLanguage)) {
+        config.preconditionLanguage = LANGUAGE_SQL;
```

Comment:
Seems like we are checking the if condition multiple times in this class. Can we move this to a common function and add some comments to make it more readable?

Reply:
We can determine and store the mode of execution in a global variable. I believe we have 2 modes
1. jexl execution mode.
2. sql engine execution mode.